### DISCUSSION TITLE
foundry devops installation command '--no-commit' added at the end.

### DISCUSSION THREAD

**Discussion by LareineHan** [Cyfrin/foundry-full-course-cu#865] [General]
Hi all, 
I don't know why so many commands aren't matching Patrick's versions. 
Anyway, here's what I did today. 

When I install foundry devops on terminal, 
This was the command on the github repo. So I tried. 
```zsh
forge install Cyfrin/foundry-devops 
```

I receive this error. 
```zsh
Error: 
The target directory is a part of or on its own an already initialized git repository,
and it requires clean working and staging areas, including no untracked files.

Check the current git repository's status with `git status`.
Then, you can track files with `git add ...` and then commit them with `git commit`,
ignore them in the `.gitignore` file, or run this command again with the `--no-commit` flag.
```

and I tried this and worked perfectly. 
```zsh
forge install Cyfrin/foundry-devops --no-commit
```

I thought some people should've faced this issue as well.

**Comment by WaleedMaghraby** [comment]
Same issue with me, ive checked git, forge, and jq versions , its OK
error message appeared:
" (node:30930) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
ℹ Detected platform is: Ubuntu
✖ Invalid Version: Cyfrin/foundry-devops
✖ Forge initialize failed, please try again later"

**Reply by WaleedMaghraby** [reply]
@LareineHan try this command
"npm install punycode --save"

**Comment by KMean** [comment]
> Hi all, I don't know why so many commands aren't matching Patrick's versions. Anyway, here's what I did today.
> 
> When I install foundry devops on terminal, This was the command on the github repo. So I tried.
> 
> ```shell
> forge install Cyfrin/foundry-devops 
> ```
> 
> I receive this error.
> 
> ```shell
> Error: 
> The target directory is a part of or on its own an already initialized git repository,
> and it requires clean working and staging areas, including no untracked files.
> 
> Check the current git repository's status with `git status`.
> Then, you can track files with `git add ...` and then commit them with `git commit`,
> ignore them in the `.gitignore` file, or run this command again with the `--no-commit` flag.
> ```
> 
> and I tried this and worked perfectly.
> 
> ```shell
> forge install Cyfrin/foundry-devops --no-commit
> ```
> 
> I thought some people should've faced this issue as well.

This is a common issue when installing dependencies in a forge project because forge initializes a Git repository by default. The error happens because forge install interacts with Git to add a submodule, but it requires a clean working directory. If you have untracked files or uncommitted changes, the installation fails.

To fix this:

Run 
```bash
git status
```
to see the current state of your Git repository.

If there are untracked or changed files:
Add and commit them:
```bash
git add .
git commit -m "Save changes before installing dependencies"
```
Ignore unwanted files by adding them to your .gitignore file if you don’t want Git to track them or remove unwanted files:
```bash
git clean -f
```
Run:
```bash
forge install Cyfrin/foundry-devops
```
If you don’t want to commit your changes yet, you can use the --no-commit flag to bypass this requirement and later add them manually to git if you want to:

```bash
forge install Cyfrin/foundry-devops --no-commit
```


If I remember correctly Patrick use the --no-commit flag, what lesson are you referring to?

**Reply by EngrPips** [reply]
Awesome.
